todoman/ui.py

```diff
@@ -70,31 +78,46 @@ def __init__(self, todo, databases, formatter):
         grid = urwid.Pile(pile_items)
         spacer = urwid.Divider()
 
-        items = [grid, spacer, buttons]
+        items = [grid, spacer, self._msg_text, buttons]
 
         self._ui = urwid.ListBox(items)
 
+    def message(self, text):
+        self._msg_text.set_text(text)
+
     def edit(self):
         """
         Shows the UI for editing a given todo. Returns True if modifications
         were saved.
         """
         self._loop = urwid.MainLoop(
             self._ui,
+            palette=_palette,
             unhandled_input=self._keypress,
             handle_mouse=False,
         )
         try:
             self._loop.run()
-        except Exception:
+        except Exception as e:
+            tb = sys.exc_info()[-1]
             try:  # Try to leave terminal in usable state
                 self._loop.stop()
             except Exception:
                 pass
+            raise e.with_traceback(tb)
```

Comment:
If stopping the loop raises an exception it would be suppressed in the inner try-except and reraised here. This is not the behavior we want, we want to reraise the original exception before we tried to stop the loop.

It appears that this has changed for Python 3 and `raise` is now somehow aware of scoped trys.